core/templates/pages/contributor-dashboard-admin-page/services/format-contribution-attributes.service.ts

```diff
@@ -0,0 +1,256 @@
+// Copyright 2024 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+/**
+ * @fileoverview Service to format attributes of contribution data for display
+  * in the contribution admin stats component expanded detail section.
+ */
+import { Injectable } from '@angular/core';
+import { QuestionReviewerStats, QuestionSubmitterStats, TranslationReviewerStats, TranslationSubmitterStats } from '../contributor-dashboard-admin-summary.model';
+import { AppConstants } from 'app.constants';
+
+export interface ContributionAttribute {
+    key: string;
+    value: string;
+  }
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FormatContributionAttributesService {
+  contributionAttribute:
+    ContributionAttribute[] = [];
+
+  private pluralise(num: number, word: string): string {
+    return num === 1 ? word : word + 's';
+  }
+
+  /**
+  * If the contribution is a review, return appropriate text and the
+  * number that have been edited.
+  * if the contribution is a submission, return appropriate text and
+  * the number that have not been edited.
+  *
+  * @param {number} numberOfEdits - The number of edits
+  * @param {number} contributionSubType - review or submission
+  * @returns {string} The text to be displayed on screen, with brackets
+  */
+  private getEditTextForContribution(
```

Comment:
You can probably just inline this within its caller, it's only used once.

Reply:
Done